algoliasearch/client.go

```diff
@@ -243,6 +243,92 @@ func (c *client) MultipleQueriesWithRequestOptions(queries []IndexedQuery, strat
 	return
 }
 
+func (c *client) ListClusters() (res []Cluster, err error) {
+	return c.ListClustersWithRequestOptions(nil)
+}
+
+func (c *client) ListClustersWithRequestOptions(opts *RequestOptions) (res []Cluster, err error) {
+	var tmp map[string][]Cluster
+	err = c.request(&tmp, "GET", "/1/clusters", nil, read, opts)
+	res = tmp["clusters"]
+	return
+}
+
+func (c *client) ListUserIDs(page int, hitsPerPage int) (res listUserIDsRes, err error) {
+	return c.ListUserIDsWithRequestOptions(page, hitsPerPage, nil)
+}
+
+func (c *client) ListUserIDsWithRequestOptions(page int, hitsPerPage int, opts *RequestOptions) (res listUserIDsRes, err error) {
+	params := Map{
+		"page":        page,
+		"hitsPerPage": hitsPerPage,
+	}
+	err = c.request(&res, "GET", "/1/clusters/mapping?"+encodeMap(params), nil, read, opts)
+	return
+}
+
+func (c *client) GetUserID(userID string) (res UserID, err error) {
+	return c.GetUserIDWithRequestOptions(userID, nil)
+}
+
+func (c *client) GetUserIDWithRequestOptions(userID string, opts *RequestOptions) (res UserID, err error) {
+	err = c.request(&res, "GET", "/1/clusters/mapping/"+url.QueryEscape(userID), nil, read, opts)
```

Comment:
Instead of directly passing the `userID` here, let's early return an error if it is empty. This way, we avoid an extra call to the engine and we can add a simple test for it.

Reply:
Yeah, agreed 👍